### ISSUE TITLE
Reaction rate of the Allen-Cahn equation data

### ISSUE THREAD

**Issue by functoreality** [camlab-ethz/poseidon#11]
I am very glad to have access to your publicly-available benchmark dataset. In Appendix B.2.12 of the paper, it is mentioned that the Allen-Cahn equation takes the form $u_t = \Delta u - \epsilon^2 u (u^2 - 1)$, with a reaction rate of $\epsilon=220$. The simulation time interval $[0,0.0002]$ is divided into 19 time steps, yielding a total of 20 evenly-spaced snapshots.

My problem is related to the reaction rate $\epsilon$ in this equation. Recently, I tried to use another solver to solve this equation under the same initial condition. Compared with the snapshots provided in the dataset I downloaded, the solutions I obtained seem to evolve faster. Therefore, I tend to conjecture that the real reaction coefficient $\epsilon^2$ used to generate this dataset could be smaller than $220^2$.
I also write a simple script to estimate this coefficient:
```python
import numpy as np
import h5py

reader = h5py.File("ACE.nc", "r")

def estimate_reaction_rate(data_idx):
    u = reader["solution"][data_idx]  # shape: [time, x, y]
    t = np.linspace(0, 2e-4, 20)
    ut = np.gradient(u, t, axis=0)
    laplace_u = 128**2 * (np.roll(u, 1, 1) + np.roll(u, -1, 1)
                          + np.roll(u, 1, 2) + np.roll(u, -1, 2) - 4 * u)
    lhs = ut - laplace_u
    reac_term = u * (u**2 - 1)
    eps2 = np.sqrt(np.mean(lhs**2) / np.mean(reac_term**2))
    return eps2

for i in range(16):
    print(i, estimate_reaction_rate(i))
```
I expect to get values around $220^2=48400$. But the output was:
```
0 14493.126
1 14949.7705
2 15157.49
3 15969.236
4 17535.186
5 17473.295
6 16101.737
7 17425.406
8 16187.239
9 14629.087
10 15235.188
11 16618.508
12 15381.288
13 15621.207
14 15965.024
15 15240.549
```
Have I misunderstood the simulation setting?

**Comment by bogdanraonic3** [comment]
Hi, the first thing that stands out is that the time step used is $dt = 0.15 \times 10^{-7}$ to ensure numerical stability. Using your time step would make the simulation unstable. 

Best,
Bogdan

**Comment by functoreality** [comment]
Well, I modified my script to estimate the time step of the simulation data (assuming $\epsilon=220$ is correct), and the result seems to be $dt\approx 3.0\times 10^{-6}$. (And indeed, I might have made some mistakes that I did not realize.)
As far as I know, a common practice in PDE data generation is to distinguish between the simulation time step and the saving time step. That is, the simulation time step is taken to be small to ensure numerical stability, and could be determined on-the-fly according to the CFL conditions. Each solution snapshot is saved until reaching the pre-specified saving time step, with the intermediate simulation steps discarded. Maybe you also followed this practice when generating this dataset?

**Comment by maximilianherde** [comment]
Exactly, there were more simulation time steps than snapshots saved.
